packages/fether-react/src/BackupPhrase/BackupPhrase.js

```diff
@@ -0,0 +1,394 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+import * as CryptoJS from 'crypto-js';
+import { AccountCard, Header, Form as FetherForm } from 'fether-ui';
+import localForage from 'localforage';
+import { inject } from 'mobx-react';
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+import RequireHealthOverlay from '../RequireHealthOverlay';
+import withAccount from '../utils/withAccount';
+
+/*
+  feat #100: The user skipped phrase rewrite during account creation.
+  Then ->
+    Two cases:
+    1. The user hasn't confirmed recovery phrase backup yet.
+      -> In this case we show a warning and guide them to decrypt the phrase with their password and through to the rewrite step.
+    2. The user has confirmed backup of their recovery phrase, but just wants to view it once more.
+      -> In this case, we do not show warning, but we alow them to decrypt the phrase with their password.
+*/
+@inject('parityStore')
+@withAccount
+class BackupPhrase extends Component {
+  state = {
+    error: null,
+    password: '',
+    phrase: '',
+    phraseRewrite: '',
+    step: 1,
+    unlocked: false
+  };
+
+  componentDidMount () {
+    this.getPhrase();
+  }
+
+  handleChangePassword = ({ target: { value } }) => {
+    this.setState({ password: value });
+  };
+
+  // handle the user typed phrase rewrite
+  handleChangePhrase = ({ target: { value } }) => {
+    this.setState({ phraseRewrite: value });
+  };
+
+  handleSubmit = async event => {
+    const {
+      account: { address },
+      history
+    } = this.props;
+    const { password, phrase } = this.state;
+
+    // unset account flag
+    await localForage.removeItem(`__flagged_${address}`);
+
+    // AES encrypt the phrase
+    const encryptedPhrase = CryptoJS.AES.encrypt(phrase, password).toString();
```

Comment:
if we're importing from a json keyfile, `phrase` will be null and I think the function will throw. we need to skip the aes encryption step if importing from a json keyfile

Reply:
Thanks for bringing this one up, just tested and on JSON import, the account will not be flagged, and therefore the account won't have the option to backup/view the recovery phrase since it won't exist, as you said.